### ISSUE TITLE
Tools: Deprecate/remove cygwin support

### ISSUE THREAD

**Issue by tracernz** [d-ronin/dRonin#1327]
Doubtful it works. https://github.com/d-ronin/dRonin/blob/2e9629b9e7d9aad48975a9a052e65c0c92fd2244/make/system-id.mk#L38

**Comment by tracernz** [comment]
Should probably do the same for MinGW GCS builds unless somebody takes on responsibility to maintain it.

**Comment by mlyle** [comment]
Can't do it for mingw unless/until we have appropriate developer documentation / streamlined setup.

**Comment by tracernz** [comment]
New developer docs worked for @yamahabest. Seem to work for me on a virgin VM too (just waiting for ARM SDK to install then I can build). https://dronin.readme.io/v99-dev/docs/development-setup-windows Used Qt online installer ATM because it's easiest, would be good to come up with an alternate long-term though.

@jihlein

**Comment by tracernz** [comment]
On my VM, corecrt.h is missing.
http://i.imgur.com/bDWxhob.png

-e- We need Windows 10 kit:
https://github.com/d-ronin/dRonin/blob/next/make/winx86/bash_profile#L24

**Comment by jihlein** [comment]
"Step 3. Expand "Windows and Web Development", then "Universal Windows App Development Tools". Select "Windows 10 SDK (10.0.10240)" (PLEASE MAKE SURE YOU GET THIS EXACT VERSION)."

"-e- We need Windows 10 kit:
https://github.com/d-ronin/dRonin/blob/next/make/winx86/bash_profile#L24"

Does this mean it only works on Windows 10?

**Comment by tracernz** [comment]
No, that screenshot is on Windows 8.1.

**Comment by jihlein** [comment]
Okay, it wasn't pretty, but I have it working, to a point.  I had to uninstall basically every tool, and then re install them, starting with GIT.  I have no idea why, something was corrupted somewhere.  Did not reinstall the compiler.

My typical build command, GCS_BUILD_CONF=release FLIGHT_BUILD_CONF=release make -j7 all make package_flight does not work, it throws an error.  It starts building the bootloaders, but then errors out on the first line of building uavos, or the GCS, don't recall which.  But GCS_BUILD_CONF=release FLIGHT_BUILD_CONF=release make -j7 gcs, followed by GCS_BUILD_CONF=release FLIGHT_BUILD_CONF=release make -j7 all_fw after the GCS build completes works.  I'll get better debug info posted later.

Three things to note:
1)The figure in step 4 of the setup should be updated.  It needs to show  the line for Windows 10 SDK (10.0.10240).  At least when I installed MSVC that line was present in the dialog box.
2)I had to add "C:\Qt\Tools\mingw492_32\bin;" to the path environment variable to be able to find make.
3)I had to add "C:\Qt\5.6\msvc2015\bin;" to the path environment variable to be able to run the compiled GCS.

Should mention that this is on Windows 7 Pro, 64 bit.

**Comment by jihlein** [comment]
Not sure what's really going on, but the only way I can get "complete" builds is to run:
GCS_BUILD_CONF=release make -j7 fw and let that complete.  Then run:
FLIGHT_BUILD_CONF=release make -j7 all_fw and let that complete.  

At that point I have a working GCS and firmware for all the targets.  Any command with a make -j7 all or make -j7 all_flight will seem to run, but randomly ends, leaving parts not built.  It does not always stop at the same point.  A make -j7 all typically builds bootloaders, bootloader updaters, tries to build some firmware, gives up on firmware for a reason I can't determine, and builds most of the GCS, again not completing.  Something about the parallel builds under MSVC??

**Comment by tracernz** [comment]
Try `make -j7 all 1>/dev/null` so the stdout noise doesn't hide any errors. I suspect there will be something in there. You could also do `make -j7 all > buildlog.txt` to redirect all the output to a file which you can post a link to here (via Gist or pastebin). I've had no issues with parallel builds, and I think the Jenkins slave does parallel builds too so it shouldn't be related to that (other than a missing dep that only bites when jobs run in a particular order..?).

**Comment by mlyle** [comment]
Jenkins only uses very mild parallelism-- I think -j2 or -j3

**Comment by mlyle** [comment]
Oh, one annoyance/weirdness with Windows-- when I ctrl-c a build, it tends to leave behind incomplete/0 byte build outputs that screw up subsequent builds.

**Comment by jihlein** [comment]
Fixed!  While outputting the build to buildlog.txt, an error appeared on the MINGW64 terminal window during the build of the bootloader updater for playuavosd.  It was executing px_mkfw.py, and actually seg faulted due to a conflict with cygwin1.dll versions.  I found some remnants of an old cygwin installation that had not been deleted, and the cygwin path variables were still in place.  Cleaned that up and it now works and generates everything from make all.  Not sure why that error didn't show before, but at least the issue is solved.  I've successfully built next, and it's now building "myNext".

Thanks for all the patience and help!!
